src/main/java/ca/corefacility/bioinformatics/irida/config/security/IridaOauthSecurityConfig.java

```diff
@@ -0,0 +1,236 @@
+import org.springframework.security.oauth2.provider.code.InMemoryAuthorizationCodeServices;
+import org.springframework.security.oauth2.provider.error.AbstractOAuth2SecurityExceptionHandler;
+import org.springframework.security.oauth2.provider.error.OAuth2AuthenticationEntryPoint;
+import org.springframework.security.oauth2.provider.error.WebResponseExceptionTranslator;
+import org.springframework.security.oauth2.provider.token.DefaultTokenServices;
+import org.springframework.security.oauth2.provider.token.ResourceServerTokenServices;
+import org.springframework.security.oauth2.provider.token.TokenStore;
+import org.springframework.security.web.context.SecurityContextPersistenceFilter;
+import org.springframework.security.web.util.matcher.AntPathRequestMatcher;
+import org.springframework.util.ReflectionUtils;
+
+import java.lang.reflect.Field;
+
+/**
+ * Configuration for REST API security using OAuth2
+ */
+@Configuration
+public class IridaOauthSecurityConfig {
+  private static final Logger logger = LoggerFactory.getLogger(IridaOauthSecurityConfig.class);
+
+  @Bean
+  @Primary
+  public ResourceServerTokenServices tokenServices(@Qualifier("clientDetails") ClientDetailsService clientDetails,
+    @Qualifier("iridaTokenStore") TokenStore tokenStore) {
+    DefaultTokenServices services = new DefaultTokenServices();
+    services.setTokenStore(tokenStore);
+    services.setSupportRefreshToken(true);
+    services.setClientDetailsService(clientDetails);
+    return services;
+  }
+
+  @Bean
+  public ClientDetailsUserDetailsService clientDetailsUserDetailsService(
+    @Qualifier("clientDetails") ClientDetailsService clientDetails) {
+    ClientDetailsUserDetailsService clientDetailsUserDetailsService = new ClientDetailsUserDetailsService(
+      clientDetails);
+
+    return clientDetailsUserDetailsService;
+  }
+
+  @Bean
+  public WebResponseExceptionTranslator exceptionTranslator() {
+    return new CustomOAuth2ExceptionTranslator();
+  }
+
+  /**
+   * Class for configuring the OAuth resource server security
+   */
+  @Configuration
+  @EnableResourceServer
+  @ComponentScan(basePackages = "ca.corefacility.bioinformatics.irida.repositories.remote")
+  @Order(Ordered.HIGHEST_PRECEDENCE + 2)
+  protected static class ResourceServerConfiguration extends ResourceServerConfigurerAdapter {
+
+    @Autowired
+    private ResourceServerTokenServices tokenServices;
+
+    @Autowired
+    private WebResponseExceptionTranslator exceptionTranslator;
+
+    @Override
+    public void configure(final ResourceServerSecurityConfigurer resources) {
+      resources.resourceId("NmlIrida").tokenServices(tokenServices);
+      forceExceptionTranslator(resources, exceptionTranslator);
+    }
+
+    @Override
+    public void configure(final HttpSecurity httpSecurity) throws Exception {
+      httpSecurity.antMatcher("/api/**").authorizeRequests()
+        //.antMatchers("/api/oauth/authorize").fullyAuthenticated()
```

Comment:
These antMatchers are commented out as they are being handled by the other WebSecurity config, see another comment below.